triveni-codev/GBWebApp-dev
# Justin_Outputs-Section_Implement-field-validations-in-IPaddress-PORT_1203813519413086

ASANA Ticket: https://app.asana.com/0/1203786645839927/1203813519413086/f

**Description:** Implement the full restrictions/validations in the IP address PORT field
- Created a special character validation
- Created a numberOnly and PORT directive to be implemented in the field
- Created a logic function which will determine if the IP Address and PORT field values are valid and not empty
- Implemented the above logic to the show/hide of the green check-mark icon
- Created a logic function which will show/hide the PORT field depends on the Output Type
- Added a default **4240** PORT value if the Output Type is "Thomson NetProcessor"
- Created a function which will disable the "Test Connection" and "UPDATE" button if the IP Address and PORT values are invalid or empty
- Added a tooltip range
- Added a max value **(65535)** validation for the PORT field


**TO TEST:**
- Create/Update an Output via dirty change or server using only the ff. Output Types (ATSC3, Carousel, Thomson, The 3 Ericssons)
- Test the IP address field validation (should only accept valid IPs)
- Test the IP PORT field validation  (should only accept numbers and 1025-65535 range)
- Test the "Test Connection" , "UPDATE" , and "ADD" button enabling/disabling logic upon entering the IP Address and PORT values
- Test also the visibility of green check-mark icons if it is accurate upon entering the IP Address and PORT values

NOTE: everything that are not mentioned in the test cases above are OUT-OF-SCOPE of this ticket/PR and it will be fixed on a separate tickets with separate PRs. Kindly disregard those issues you might encountered, but feel free to log it as it will help me to track all the bugs/issues in the Output Section. Thank you.

cc: @triveni-codev
